### DISCUSSION TITLE
Can't deploy anything. Event a very basic project with a few packages, gives me npm ERROR 401: Incorrect credentials

### DISCUSSION THREAD

**Discussion by albertnev** [vercel/community#7237] [Help] [answered]
### Summary

So basically, [I wrote this thread the other day](https://github.com/orgs/vercel/discussions/7218) which got no answers, but I thought at the end this had to be something I'm doing wrong, so I've kept investigating. However, today I've tried to deploy a very simple project, an empty project with just some basic packages installed, and I'm still getting the same error.

Is there some kind of limit on how many projects I can deploy (I already have one that's deployed)? I don't understand how I'm unable to deploy anything, even if I set an NPM_TOKEN env variable.

Any help would be much appreciated.

### Steps to Reproduce

Just create a simple Github project, import it and try to deploy it. It will fail in the "installing packages" phase. The package.json is the following:

```
{
  "name": "project-draft",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "test": "vitest run",
    "e2e": "npx playwright test",
    "e2e-show": "npx playwright show-report"
  },
  "dependencies": {
    "@types/jest": "^29.5.12",
    "clsx": "^2.1.1",
    "next": "14.2.3",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@next/eslint-plugin-next": "^14.2.3",
    "@playwright/test": "^1.44.1",
    "@testing-library/jest-dom": "^6.4.5",
    "@testing-library/react": "^16.0.0",
    "@types/node": "^20.14.2",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "@vercel/style-guide": "^6.0.0",
    "@vitejs/plugin-react": "^4.3.1",
    "eslint": "^8.2.0",
    "eslint-config-next": "14.2.3",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-sort-destructure-keys": "^2.0.0",
    "eslint-plugin-typescript-sort-keys": "^3.2.0",
    "jsdom": "^24.1.0",
    "postcss": "^8",
    "prettier": "^3.3.1",
    "tailwindcss": "^3.4.1",
    "tailwindcss-animate": "^1.0.7",
    "typescript": "^5",
    "vitest": "^1.6.0"
  }
}
```

**Comment by swarnava** [comment]
Hi,

-   Try Deleting the package-lock.json file before running the command "npm install" after upgrading the node versions, if you have not done that.

- If that will not work try removing the .npmrc file in $HOME directory as mentioned in this post.

[NPM ERR Code E401: Unable to authenticate, need: Bearer authorization](https://stackoverflow.com/questions/59894644/npm-err-code-e401-unable-to-authenticate-need-bearer-authorization)

Note - Deleting package-lock.json can have its own issues as mentioned in this posts, so go with this solution unless you find any better solution.

[Deleting `package-lock.json` to Resolve Conflicts quickly](https://stackoverflow.com/questions/54124033/deleting-package-lock-json-to-resolve-conflicts-quickly?noredirect=1&lq=1)

**Reply by albertnev** [reply]
Hello, and thanks for your reply! 

Unfortunatelly, **I've got no `.npmrc` file in my project** because all of the packages I'm referencing are in the default public npm registry, so I didn't need it at all. Also, **I've already tried to delete multiple times the `package-lock.json` file**, including deleting the whole `node-modules` (not that it makes any effect, but just to avoid any cache related thing), and still no luck. **I keep getting the same error.**

That's why I'm so lost, because there's no more explanation than the 401 error, but all the packages are public, so I don't know why I could be getting that error.

Thanks.

**Reply by albertnev** [reply, marked as answer]
However, you gave me an idea, and that was deleting completely the `package-lock.json` file, and uploading the commit with the file deleted, and this seems to have worked. I don't understand why, and I know that could give me problems (since each time I make a deployment, I'm at risk that some version update could mess with my application), but that's better than not being able to deploy it. Thank you so much!

[a comment by an automated account is not shown]
